src/test/java/ReceiptAnalysisTest.java

```diff
@@ -83,4 +97,72 @@ public void serveImageTextUrlReturnsAnalysisResults() throws IOException {
     Assert.assertEquals(results.getRawText(), RAW_TEXT);
     verify(client).batchAnnotateImages(requests);
   }
+
+  @Test
+  public void serveImageTextThrowsIfEmptyBatchResponse()
+      throws IOException, ReceiptAnalysisException {
+    URL url = mock(URL.class);
+    InputStream inputStream = new ByteArrayInputStream(IMAGE_BYTES.toByteArray());
+    when(url.openStream()).thenReturn(inputStream);
+
+    ImageAnnotatorClient client = mock(ImageAnnotatorClient.class);
+    mockStatic(ImageAnnotatorClient.class);
+    when(ImageAnnotatorClient.create()).thenReturn(client);
+
+    BatchAnnotateImagesResponse batchResponse = BatchAnnotateImagesResponse.newBuilder().build();
+    when(client.batchAnnotateImages(Mockito.<AnnotateImageRequest>anyList()))
+        .thenReturn(batchResponse);
+
+    expectedException.expect(ReceiptAnalysisException.class);
+    expectedException.expectMessage(EMPTY_BATCH_RESPONSE_WARNING);
```

Comment:
For testing exceptions, I recommend using assertThrows(). You want to break your test up into three parts: arrange, act, and assert. You may need to upgrade your version of junit for this.

https://stackoverflow.com/questions/40268446/junit-5-how-to-assert-an-exception-is-thrown
https://howtodoinjava.com/junit5/expected-exception-example/